### PR TITLE
Allow ETH transfer on the Sent Asset modal

### PR DESCRIPTION
**Pull request by mobiusTripper-crypto**
## Description
Added the ability to sent native ETH using the Send Assets Modal.

This incorporates (https://github.com/decent-dao/fractal-interface/issues/754) with the addition of the input component BigNumberInput. Upon this PRs testing and completion, this component can be use throughout the rest app where other big numbers are used.

## Notes
Disable USD switch and created issue to re-enable with correct functionality.
No toast / notification on transaction submit, created issue to add.

## Issue / Notion doc (if applicable)
https://github.com/decent-dao/fractal-interface/issues/810

## Testing
From the Treasury page. Insure DAO has ETH in the treasury and the current user is a delegate or owner.
Open the Send Assets modal
Test all possible combination of input on the token amount input component, i.e. leading zeros, non number characters, copy / paste, etc.
Create a proposal to send native ETH using the Send Assets modal, and verify the ETH is transfer once the proposal is executed.
 
## Screenshots (if applicable)
image

### CODE REVIEW

File: src/components/ui/BigNumberInput/BigNumberInput.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      });
+      if (newValue !== event.target.value) {
+        resetCaretPositionForInput(event);
+      }
+      setInputValue(newValue);
+    },
+    [decimalPlaces, onChange, truncateDecimalPlaces]
+  );
+
+  // if the decimalPlaces change, need to update the value
```
Comment (mobiusTripper-crypto) [inline src/components/ui/BigNumberInput/BigNumberInput.tsx:87]: When the asset is changed on the modal, we need to re-run the truncateDecimalPlaces. I don't feel like my use of the useEffect here is 100% correct. But I couldn't figure out how to trigger onChange on the input?

**Review by Da-Colon — APPROVED** [review]
I won't hold PR, if you want to just do it in another one.

File: src/components/ui/modals/SendAssetsModal.tsx [hunk 2]
```diff
@@ -1,105 +1,71 @@
...
 
-  const isSubmitDisabled =
-    isEth(selectedAsset) || // remove this check once we are able to send ETH
-    !isValidAddress ||
-    !amountInput ||
-    overDraft;
+  const isSubmitDisabled = !isValidAddress || inputAmount?.bigNumberValue.isZero() || overDraft;
 
   const onSubmit = () => {
     sendAssets();
```
Comment (Da-Colon) [inline src/components/ui/modals/SendAssetsModal.tsx:67]: Might be a quick add to go ahead and add toasts here using `useTransaction()` hook. see `src/components/Proposals/MultisigProposalDetails/TxActions.tsx`. its pretty straightforward

File: src/components/ui/BigNumberInput/BigNumberInput.tsx (hunk 1 above)
Comment (Da-Colon) [reply src/components/ui/BigNumberInput/BigNumberInput.tsx:87]: doesn't appear to effect the 'controlled' or 'uncontrolled' state of the Input. in my book this looks good and works well

File: src/pages/Treasury/hooks/useSendAssets.ts [hunk 3]
```diff
@@ -16,12 +16,13 @@ const useSendAssets = ({
   const { submitProposal } = useSubmitProposal();
 
   const successCallback = () => {
-    //TODO: need to add a toast and close modal
+    //TODO: need to figure out what toast / notification to add
```
Comment (herbig) [inline src/pages/Treasury/hooks/useSendAssets.ts:19]: @Da-Colon mentioned there is a `useTransaction` hook that can manage pending / success toasts, but we can add that as a followup

File: src/components/ui/modals/SendAssetsModal.tsx (hunk 2 above)
Comment (herbig) [reply src/components/ui/modals/SendAssetsModal.tsx:67]: oh just noticed this comment, yeah could add in this PR or a followup

**Review by herbig — CHANGES_REQUESTED** [review]
@mobiusTripper-crypto the USD slider no longer works, it's permanently disabled now for some reason

**Comment by mobiusTripper-crypto** [comment]
> @mobiusTripper-crypto the USD slider no longer works, it's permanently disabled now for some reason

Yes, I think it needs to switch between the token amount and the USD value. the way it works now it just confusing (and I think doesn't actually work). I am creating a follow up issue.

**Review by herbig — APPROVED** [review]

File: src/pages/Treasury/hooks/useSendAssets.ts [hunk 4]
```diff
@@ -31,16 +32,16 @@ const useSendAssets = ({
     ];
 
     const proposalData: ProposalExecuteData = {
-      targets: [tokenAddress],
-      values: [BigNumber.from('0')],
-      calldatas: calldatas,
+      targets: [isEth ? destinationAddress : asset.tokenAddress],
+      values: [isEth ? transferAmount : BigNumber.from('0')],
+      calldatas: isEth ? ['0x'] : calldatas,
       title: 'Send Assets',
```
Comment (herbig) [inline src/pages/Treasury/hooks/useSendAssets.ts:38]: just a note for @decent-dao/fractal, any on chain metadata we submit in the background (title/description) should be hardcoded in English, as this is.

translating these per the user's settings doesn't make sense, as they are not necessarily submitting it to a DAO that is specific to their language

**Pull request merged**